src/storage/storage.cpp

```diff
@@ -184,12 +184,78 @@ Storage::ReturnCode Storage::Run(int max_wait)
     }
 
     // Allocate a memory layout in shared memory
-    auto layout_memory = makeSharedMemory(layout_region, sizeof(SharedDataLayout), true);
-    auto shared_layout_ptr = new (layout_memory->Ptr()) SharedDataLayout();
+    auto layout_memory = makeSharedMemory(layout_region, sizeof(DataLayout), true);
```

Comment:
The following set-up routines are local to the SharedMemoryFacade, are they? Could we potentially encapsulate this in the Facade/Layout or provide a factory? It seems dangerous to me to require this kind of knowledge outside of the respective facade.